queue/queue.go

```diff
@@ -6,15 +6,28 @@ import (
 	"time"
 
 	"github.com/frain-dev/convoy"
+	"github.com/frain-dev/convoy/config"
 	"github.com/frain-dev/convoy/datastore"
+	"github.com/go-redis/redis/v8"
+	"github.com/vmihailenco/taskq/v3"
 )
 
 type Queuer interface {
 	io.Closer
 	Write(context.Context, convoy.TaskName, *datastore.EventDelivery, time.Duration) error
 }
 
+type QueueClient interface {
```

Comment:
@ogbanugot I don't think we want to do this. What we want to do is:
- Create the client.
- Pass it to the `NewQueue` function for the respective queue.
 
This is implementing standard DI.